### PR TITLE
06688 pces emergency reconnect3

### LINKED ISSUE #6688: Delete PCES during emergency reconnect
If we ever do an emergency reconnect, we should completely clear all data in the preconsensus event stream.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #6688

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 1]
```diff
@@ -993,9 +986,21 @@ private EventLinker buildEventLinker(@NonNull final List<Predicate<EventDescript
      * Build the preconsensus event file manager.
      */
     @NonNull
-    private PreconsensusEventFileManager buildPreconsensusEventFileManager() {
+    private PreconsensusEventFileManager buildPreconsensusEventFileManager(
+            final EmergencyRecoveryManager emergencyRecoveryManager) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:990]: Nullity annotation

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 2]
```diff
@@ -993,9 +986,21 @@ private EventLinker buildEventLinker(@NonNull final List<Predicate<EventDescript
...
         try {
-            return new PreconsensusEventFileManager(platformContext, OSTime.getInstance(), recycleBin, selfId);
+            final PreconsensusEventFileManager manager =
+                    new PreconsensusEventFileManager(platformContext, OSTime.getInstance(), recycleBin, selfId);
+
+            if (emergencyRecoveryManager.isEmergencyRecoveryFilePresent()) {
+                logger.info(
+                        STARTUP.getMarker(),
+                        "This node was started in emergency recovery mode, "
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:998]: Nit: unnecessary line break in string

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:998]: Technically, yes. But it bothers my OCD when the intellij hints push something over the line count boundary.

Screenshot 2023-06-15 at 1 01 45 PM

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:990]: added

**Review by alittley — APPROVED** [review]

**Pull request merged**